core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -562,13 +562,17 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
             // /config/users/<default>/clients/<default>
             quota = overriddenQuotas.get(DefaultUserClientIdQuotaEntity)
           }
+          if (quota == null)
+            quota = ClientQuotaManagerConfig.UnlimitedQuota
```

Comment:
With this change, it seems like we don't need to let this function return null anymore. In all the branches, the final result will be `ClientQuotaManagerConfig.UnlimitedQuota` or `staticConfigClientIdQuota`. That would let us simplify some of the usages.

Reply:
There is also a case where `metricTags` has no `user` or `client-id` keys. Based on the description of `quotaLimit` method in `ClientQuotaCallback`, `null` is a valid return value if no metrics tags in use. @rajinisivaram Do you know what exactly "if the metric tags are no longer in use" means in the description. Does it cover the case where `metricTags` is empty for example or does not have expected tags? My understanding is that it does.

```
     * @return the quota limit for the provided metric tags or null if the metric tags are no longer in use
     */
    Double quotaLimit(ClientQuotaType quotaType, Map<String, String> metricTags);
```

Reply:
I see. So we cannot avoid checking the return type for null. I guess at least we can clean up this function though.